### PR TITLE
#29 Adding ForceNew flag for template_id field

### PR DESCRIPTION
**Pull request by ja6a**
Adding ForceNew flag for template_id field. This will ensure changing a template id on an existing environment will cause a new one to be created.

### CODE REVIEW

File: skytap/resource_skytap_environment.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func resourceSkytapEnvironment() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validation.NoZeroValues,
+				ForceNew: true,
```
Comment (pegerto) [inline skytap/resource_skytap_environment.go]: we probably need acceptance test on this

File: skytap/resource_skytap_environment_test.go [hunk 2]
```diff
@@ -73,8 +75,37 @@ func TestAccSkytapEnvironment_Basic(t *testing.T) {
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSkytapEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSkytapEnvironmentConfig_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSkytapEnvironmentExists("skytap_environment.foo", &environment, false),
+					resource.TestCheckResourceAttr("skytap_environment.foo", "name", fmt.Sprintf("tftest-environment-%d", rInt)),
+					resource.TestCheckResourceAttr("skytap_environment.foo", "description", "This is an environment created by the skytap terraform provider acceptance test"),
```
[the remark this answers is not in the document]
Comment (ja6a) [reply skytap/resource_skytap_environment_test.go]: Removed checks on name and description. The only the thing that concerns us in this case is that the environment resource id changes.

File: skytap/resource_skytap_environment_test.go [hunk 3]
```diff
@@ -73,8 +75,37 @@ func TestAccSkytapEnvironment_Basic(t *testing.T) {
 	})
 }
 
+func TestAccSkytapEnvironment_UpdateTemplate(t *testing.T) {
+	rInt := acctest.RandInt()
```
[the remark this answers is not in the document]
Comment (ja6a) [reply skytap/resource_skytap_environment_test.go]: Renamed to uniqueSuffix

File: skytap/resource_skytap_environment_test.go [hunk 4]
```diff
@@ -140,3 +179,12 @@ resource "skytap_environment" "foo" {
...
 }
+
+func testAccSkytapEnvironmentConfig_update(rInt int) string {
+	return fmt.Sprintf(`
+resource "skytap_environment" "foo" {
+	template_id = "1442921"
+	name = "tftest-environment-%d"
+	description = "This is an environment created by the skytap terraform provider acceptance test"
+}`, rInt)
+}
```
[the remark this answers is not in the document]
Comment (ja6a) [reply skytap/resource_skytap_environment_test.go]: Now passing in template id and reusing the testAccSkytapEnvironmentConfig_basic method.

File: skytap/resource_skytap_environment_test.go [hunk 5]
```diff
@@ -73,8 +75,37 @@ func TestAccSkytapEnvironment_Basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr("skytap_environment.foo", "description", "This is an environment created by the skytap terraform provider acceptance test"),
+				),
+			},
+		},
+	})
+}
+
 // Verifies the Environment exists
-func testAccCheckSkytapEnvironmentExists(name string) resource.TestCheckFunc {
+func testAccCheckSkytapEnvironmentExists(name string, environment *skytap.Environment, update bool) resource.TestCheckFunc {
```
[the remark this answers is not in the document]
Comment (ja6a) [reply skytap/resource_skytap_environment_test.go]: Now using two discrete methods for the create and update template tests. Each then calls out to two functions to perform duplicated logic thereby promoting code reuse.

**Pull request merged**